### DISCUSSION TITLE
Swiper Grid, Auto SlidesPerView & rows?

### DISCUSSION THREAD

**Discussion by ivanjeremic** [nolimits4web/swiper#5661] [Q&A]
I'm very interested in using Swiper Grid Example(react) for my App, I forked the Example from codesandbox [https://codesandbox.io/s/swiper-slides-grid-react-forked-2c0041?file=/src/App.jsx](https://codesandbox.io/s/swiper-slides-grid-react-forked-2c0041?file=/src/App.jsx) and played around and I need to tweak a few things which is harder than I thought, The first problem I want to solve is the `Swiper ` decides how many `SwiperSlide`'s to show on one page based on how many `slidesPerView` there are, what I want is to calculate the `slidesPerView` dynamically based on the current size of the window, that means if currently 6 slides can fit on one page there should be less pages and when the user makes the window smaller or is on mobile there should be more pages.

The second problem which is the same but for `rows` is that the slides change the width & height and become bigger when the window width or height changes, what I want is perfectly square size `SwiperSlide`'s the size of the slide should be calculated based on how many rows/slidesPerView there could be but other than in the example the slides should stay on top no matter what height or width I give to the `Swiper` itself and `flex-wrap` when more slides are added.

**Comment by izhar360** [comment]
same issue did you find any solution? to the dynamic slidesperview based on width of screen?
